charts/xp-management/templates/deployment.yaml

```diff
@@ -40,11 +40,13 @@ spec:
         image: "{{ .Values.deployment.image.registry }}/{{ .Values.deployment.image.repository }}:{{ .Values.deployment.image.tag | default .Chart.AppVersion }}"
         imagePullPolicy: {{ .Values.deployment.image.pullPolicy }}
         env:
+          {{- if and .Values.externalPostgresql.enabled .Values.externalPostgresql.createSecret }}
```

Comment:
Hmm.. How is this working for Turing, then? https://github.com/caraml-dev/helm-charts/blob/main/charts/turing/templates/turing-deployment.yaml#L70

Reply:
Hmm for some reason there's already a `turing-turing-external-postgresql` secret lying somewhere. I'm not sure if this was created beforehand (and meant to persist) or if it was spun up by the chart, seems like the purpose of this env var is to accommodate the use of a db secret created beforehand?

Reply:
Ahh I think we missed something around this. Secret creation should be controlled ONLY by the `createSecret` config (whether or not an existing external DB is used or a new one created as a part of the helm chart). But if this flag is set to false, I think we assume the secret's name supplied is existing (related to what Maksym pointed out as well).